contribs/gmf/apps/desktop/default.js

```diff
@@ -1,12 +1,27 @@
+var urlElements = window.location.pathname.split('/');
+urlElements.pop();
+urlElements.pop();
+urlElements.pop();
```

Comment:
I looks like you are programming in [Logo](https://en.wikipedia.org/wiki/Logo_%28programming_language%29).. ;)

This could be:

    var gmfUrlElements = urlElements.slice(0, urlElements.length - 3);